File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {

        int l=0;
        int h=nums.length-1;
        int ans=-1;

        while(l<=h){
            int mid=(l+h)/2;

            if(nums[mid]==target){
                ans=mid;
                break;
            }
            
            if(nums[mid]<target){
                l=mid+1;
            }
            else{
                h=mid-1;
            }
        }

        return ans;


        
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {

        int m=matrix.size();
        int n=matrix[0].size();

        int l=0;
        int h=m*n-1;
        bool ans=false;

        int r;
        int c;

        while(l<=h){

            int mid=(l+h)/2;

            r=mid/n;
            c=mid%n;

            if(matrix[r][c]==target){
                ans=true;
                break;
            }
            else if(matrix[r][c]<target){
                l=mid+1;
            }
            else{
                h=mid-1;
            }

        }

        return ans;
        
    }
};
